typescript/schedules.md

```diff
@@ -0,0 +1,1257 @@
+  /**
+   * Schedule Id
+   *
+   * We recommend using a meaningful business identifier.
+   */
+  id: string;
+
+  /**
+   * Additional non-indexed information attached to the Schedule. The values can be anything that is
+   * serializable by the {@link DataConverter}.
+   */
+  memo?: Record<string, any>;
+
+  /**
+   * Additional indexed information attached to the Schedule. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom Data Converter is provided.
+   */
+  searchAttributes: SearchAttributes;
+
+  // TODO flatten this?
+  info: ScheduleInfo;
+}
+
+/**
+ * The current Schedule details. They may not match the Schedule as created because:
+ * - some fields in the state are modified automatically
+ * - the schedule may have been modified by {@link ScheduleHandle.update} or
+ *   {@link ScheduleHandle.pause}/{@link ScheduleHandle.unpause}
+ */
+export type ScheduleDescription = Schedule & {
+  /** When Actions should be taken */
+  spec: RequireAtLeastOne<ScheduleSpec, 'calendars' | 'intervals'>;
+
+  /**
+   * Which Action to take
+   */
+  action: ScheduleActionOptions;
+
+  /**
+   * Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+   * running.
+   *
+   * @default {@link ScheduleOverlapPolicy.Skip}
+   */
+  overlap: ScheduleOverlapPolicy;
+
+  /**
+   * The Temporal Server might be down or unavailable at the time when a Schedule should take an Action. When the Server
+   * comes back up, `catchupWindow` controls which missed Actions should be taken at that point. The default is one
+   * minute, which means that the Schedule attempts to take any Actions that wouldn't be more than one minute late. It
+   * takes those Actions according to the {@link ScheduleOverlapPolicy}. An outage that lasts longer than the Catchup
+   * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+   *
+   * @default 1 minute
+   */
+  catchupWindow: Milliseconds;
+
+  /**
+   * When an Action times out or reaches the end of its Retry Policy, {@link pause}.
+   *
+   * With {@link ScheduleOverlapPolicy.AllowAll}, this pause might not apply to the next Action, because the next Action
+   * might have already started previous to the failed one finishing. Pausing applies only to Actions that are scheduled
+   * to start after the failed one finishes.
+   *
+   * @default false
+   */
+  pauseOnFailure: boolean;
+
+  /**
+   * Informative human-readable message with contextual notes, e.g. the reason
+   * a Schedule is paused. The system may overwrite this message on certain
+   * conditions, e.g. when pause-on-failure happens.
+   */
+  note?: string;
+
+  /**
+   * Is currently paused.
+   *
+   * @default false
+   */
+  paused: boolean;
+
+  /** Whether the number of Actions to take is limited. */
+  actionsAreLimited: boolean;
```

Comment:
the separate bool in the proto is because of a proto limitation (can't distinguish zero from missing). if you can make that distinction in ts/js, maybe better to do that and drop this.

Reply:
I like the readability of

```
if (schedule.actionsAreLimited)
```

versus

```
if (schedule.remainingActions === undefined)
```

Reply:
```
if (schedule.remainingActions === undefined)
```